crates/sdk/src/env.rs

```diff
@@ -31,6 +31,18 @@ fn expected_boolean<T>(e: u32) -> T {
     panic_str(&format!("Expected 0|1. Got {e}"));
 }
 
+pub fn get_executor_identity() -> String {
+    unsafe { sys::get_executor_identity(DATA_REGISTER) };
+    read_register(DATA_REGISTER)
+        .map(|data| String::from_utf8(data).expect("Invalid UTF-8"))
```

Comment:
if it can fail, return `Result` or `Option`, defaulting doesn't make sense here

or terminate the runtime if it's absence doesn't make sense (in this case, it doesn't) the execution must always have an associated identity.

returning a string also doesn't work here, it should be a `[u8; 32]` instead

Reply:
I've adjusted the return type - originally I had implemented the identity using `String`, as it wasn't clear until later that it would be a public key, and when I updated that I missed this location. Good spot! 👍